osc-server/src/main/java/org/osc/core/broker/util/ServerUtil.java

```diff
@@ -480,230 +472,6 @@ public void run() {
         return new Thread(timeMonitorThread, "Time-Monitor-Thread");
     }
 
-    /**
-     * Upgrade server.
-     *
-     * @param upgradeFile
-     * @throws IOException
-     */
-    public static void upgradeServer(File upgradeFile) throws IOException {
-        String upgradeFilePath = upgradeFile.getCanonicalPath();
-        if (!FilenameUtils.getExtension(upgradeFilePath).equals("zip")) {
-            throw new IOException("Failed to validate file: " + upgradeFile.getName() + ".");
-        }
-
-        final String mainJar = "osc-control.jar";
-        final String tmpUpgrade = "/tmp/serverUpgradeBundle";
-        final String jarFilePath = tmpUpgrade + "/opt/vmidc/bin/" + mainJar;
-
-        execWithLog("unzip -q -o -K -a " + upgradeFilePath + " -d " + tmpUpgrade);
-
-        if (!ServerUtil.validateUpgradeBundle(jarFilePath)) {
-            execWithLog("rm -rf " + tmpUpgrade);
-            throw new IOException("Failed to validate jar file: " + upgradeFile.getName() + ".");
-        }
-
-        final File serverJar = new File(mainJar);
-        final File backupServerJar = new File(mainJar + ".backup");
-        final String currentPid = getCurrentPid();
-        boolean successRename = false;
-
-        try {
-
-            // Temporary rename existing file to make room for new file
-            log.info("Upgrade (pid:" + currentPid + "): Renaming current file.");
-            backupServerJar.delete();
-            successRename = serverJar.renameTo(backupServerJar);
-            if (!successRename) {
-                // File was not successfully renamed
-                throw new IOException("Fail to rename original server file.");
-            }
-
-            // invoke the upgrade script
-            Version currentVersion = VersionUtil.getVersion();
-            String script = tmpUpgrade + "/opt/vmidc/bin/scripts/upgrade.sh";
```

Comment:
this script needs to be remove...not sure if its in 2 places.

Reply:
yes @arvindn05, I have deleted it now, It is located in only one place. @arvindn05, @sudhirappaji  Please review